app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,111 @@
+class TweetsController < ApplicationController
+
+    before_action :set_tweet, only: [:update, :stats, :like, :unlike, :bookmark, :unbookmark, :retweet, :quote]
+
+    # GET /users/:user_id/tweets
+    def index
+        @tweets = Tweet.all
+
+        respond_to do |format|
+            format.json { render json: @tweets }
+        end
+    end
+
+    # POST /tweets === tweets_path 
+    def create
+        @tweet = Tweet.new(tweet_params)
+    
+        if @tweet.save
+            tweet_hash = {
+                "id": @tweet.id,
+                "user": @tweet.user_id,
+                "content": @tweet.content,
+                "retweet_id": @tweet.retweet_id,
+                "quote_id": @tweet.quote_id,
+                "created_at": @tweet.created_at,
+                "updated_at": @tweet.updated_at
+            }
+            render json: { tweet: tweet_hash }, status: :created
+          else
+            render json: @tweet.errors, status: :unprocessable_entity
+        end
+    end
+
+
+    # PATCH/PUT /tweets/:id
+    def update
+      @tweet = Tweet.find(params[:id])
+      if @tweet.update(tweet_params)
+        tweet_hash = {
+                "id": @tweet.id,
+                "user": @tweet.user_id,
+                "content": @tweet.content,
+                "retweet_id": @tweet.retweet_id,
+                "quote_id": @tweet.quote_id,
+                "created_at": @tweet.created_at,
+                "updated_at": @tweet.updated_at
+            }
+        render json: { tweet: tweet_hash }, status: :ok
+      else
+        render json: @tweet.errors, status: :unprocessable_entity
+      end
+    end
+
+
+    # POST /tweets/:id/quote
+    def quote
+      original_tweet = Tweet.find(params[:id])
+      current_user = tweet_params[:user_id]
+  
+      @quote_tweet = Tweet.new(content: tweet_params[:content], user_id: current_user, quote_id: original_tweet.id)
+  
+      if @quote_tweet.save
+        quote_hash = {
+                "id": @quote_tweet.id,
+                "user": @quote_tweet.user_id,
+                "content": @quote_tweet.content,
+                "retweet_id": @quote_tweet.retweet_id,
+                "quote_id": @quote_tweet.quote_id,
+                "created_at": @quote_tweet.created_at,
+                "updated_at": @quote_tweet.updated_at
+            }
+        render json: { tweet: quote_hash }, status: :created
+      else
+        render json: @tweet.errors, status: :unprocessable_entity
+      end
+    end
+
+    # POST /tweets/:id/retweet
+    def retweet
+        original_tweet = Tweet.find(params[:id])
+        current_user = tweet_params[:user_id]
```

Comment:
tweet params does not have user_id included in it as it is described in your failed tests bellow, if you print out your params object you'll realized that the user id is not coming in your route `params => {"controller"=>"tweets", "action"=>"retweet", "id"=>"67", "format"=>"80"}
`

```
1) Tweets POST /tweets/:id/retweet with valid params returns 201 status response for creating a retweet
     Failure/Error: params.require(:tweet).permit(:content, :user_id)
     
     ActionController::ParameterMissing:
       param is missing or the value is empty: tweet
     # ./app/controllers/tweets_controller.rb:109:in `tweet_params'
     # ./app/controllers/tweets_controller.rb:81:in `retweet'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/actiontext-7.0.6/lib/action_text/rendering.rb:20:in `with_renderer'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/actiontext-7.0.6/lib/action_text/engine.rb:69:in `block (4 levels) in <class:Engine>'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/tempfile_reaper.rb:15:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/etag.rb:27:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/conditional_get.rb:40:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/head.rb:12:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/session/abstract/id.rb:266:in `context'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/session/abstract/id.rb:260:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:40:in `call_app'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:25:in `block in call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:25:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/method_override.rb:24:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/runtime.rb:22:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/sendfile.rb:110:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/engine.rb:530:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-test-2.1.0/lib/rack/test.rb:360:in `process_request'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-test-2.1.0/lib/rack/test.rb:153:in `request'
     # ./spec/requests/tweets_request_spec.rb:103:in `block (4 levels) in <top (required)>'

  2) Tweets POST /tweets/:id/retweet with valid params should create a retweet matching json schema
     Failure/Error: params.require(:tweet).permit(:content, :user_id)
     
     ActionController::ParameterMissing:
       param is missing or the value is empty: tweet
     # ./app/controllers/tweets_controller.rb:109:in `tweet_params'
     # ./app/controllers/tweets_controller.rb:81:in `retweet'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/actiontext-7.0.6/lib/action_text/rendering.rb:20:in `with_renderer'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/actiontext-7.0.6/lib/action_text/engine.rb:69:in `block (4 levels) in <class:Engine>'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/tempfile_reaper.rb:15:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/etag.rb:27:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/conditional_get.rb:40:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/head.rb:12:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/session/abstract/id.rb:266:in `context'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/session/abstract/id.rb:260:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:40:in `call_app'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:25:in `block in call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:25:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/method_override.rb:24:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/runtime.rb:22:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/sendfile.rb:110:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/engine.rb:530:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-test-2.1.0/lib/rack/test.rb:360:in `process_request'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-test-2.1.0/lib/rack/test.rb:153:in `request'
     # ./spec/requests/tweets_request_spec.rb:109:in `block (4 levels) in <top (required)>'

  3) Tweets POST /tweets/:id/retweet with invalid params does not retweet again
     Failure/Error: params.require(:tweet).permit(:content, :user_id)
     
     ActionController::ParameterMissing:
       param is missing or the value is empty: tweet
     # ./app/controllers/tweets_controller.rb:109:in `tweet_params'
     # ./app/controllers/tweets_controller.rb:81:in `retweet'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/actiontext-7.0.6/lib/action_text/rendering.rb:20:in `with_renderer'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/actiontext-7.0.6/lib/action_text/engine.rb:69:in `block (4 levels) in <class:Engine>'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/tempfile_reaper.rb:15:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/etag.rb:27:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/conditional_get.rb:40:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/head.rb:12:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/session/abstract/id.rb:266:in `context'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/session/abstract/id.rb:260:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:40:in `call_app'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:25:in `block in call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:25:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/method_override.rb:24:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/runtime.rb:22:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/sendfile.rb:110:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/engine.rb:530:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-test-2.1.0/lib/rack/test.rb:360:in `process_request'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-test-2.1.0/lib/rack/test.rb:153:in `request'
     # ./spec/requests/tweets_request_spec.rb:117:in `block (4 levels) in <top (required)>'

Finished in 0.1328 seconds (files took 0.45274 seconds to load)
3 examples, 3 failures
```